### PR TITLE
Config: Remove unused `manage/plugins/compatibility-warning` flag

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

This PR cleans up the `manage/plugins/compatibility-warning` feature flag that is disabled everywhere.

#### Testing instructions

* Go to `/plugins/contact-form-7?flags=-marketplace-v0.5`
* Verify plugin page still loads well.

### CODE REVIEW

File: client/my-sites/plugins/plugin-meta/index.jsx [hunk 1]
```diff
@@ -412,26 +397,6 @@ export class PluginMeta extends Component {
 		}
 	}
 
-	isVersionCompatible() {
```
Comment (jsnajdr) [inline client/my-sites/plugins/plugin-meta/index.jsx:415]: Skimming over this component, I noticed you can also remove the unused `getPlan` method. What caught my attention is that it calls `selectedSite.plan()`, but `.plan` is not a function.

Comment (tyxla) [reply client/my-sites/plugins/plugin-meta/index.jsx:415]: Ha - good catch! Removed in f7c321e431ba5d208f6d5136a5d1b5d2dfbc24fa

**Review by jsnajdr — APPROVED** [review]
:+1:

**Comment by jsnajdr** [comment]
Another funny thing about the `PluginMeta` component is that it takes a `selectedSite` prop from outside, and actually uses the value, but at the same time it calls the `getSelectedSite( state )` selector, too. Components should be consistent about where they get such info from.

**Comment by tyxla** [comment]
> Another funny thing about the `PluginMeta` component is that it takes a `selectedSite` prop from outside, and actually uses the value, but at the same time it calls the `getSelectedSite( state )` selector, too. Components should be consistent about where they get such info from.

Plugins code is a messy area that contains code from all eras of data fetching and I agree it could use some additional cleanup at some time.

**Pull request merged**